### ISSUE TITLE
SetTheme() behaves weird for widget.Form

### ISSUE THREAD

**Issue by Bekassyl** [fyne-io/fyne#842]
### Describe the bug:
Theme is changed for forms with delay, seems like sometimes delays are infinite (maybe for long forms).

### To Reproduce:
Steps to reproduce the behaviour:
1. Run the code below

### Screenshots:
![Capture](https://user-images.githubusercontent.com/24650439/78936680-aa4a1c00-7a74-11ea-829f-be9526dbb31c.JPG)

### Example code:

```go
package main
import(
	"fyne.io/fyne"
	"fyne.io/fyne/theme"
	"fyne.io/fyne/app"
	"fyne.io/fyne/widget"
)

func main(){
	myApp := app.New()
	myWindow := myApp.NewWindow("Rick&Morty")
	myWindow.SetMainMenu(fyne.NewMainMenu(fyne.NewMenu("Menu",
		fyne.NewMenuItem("Dark Theme", func() { myApp.Settings().SetTheme(theme.DarkTheme()) }),
		fyne.NewMenuItem("Light Theme", func() {myApp.Settings().SetTheme(theme.LightTheme()) }),
	)))

	form := widget.NewForm()
	form.Append("ID", widget.NewLabel("Character's Name"))
	form.Append("be", widget.NewLabel("bebe"))
	
	myWindow.SetContent(form)
	myWindow.ShowAndRun()
}
```

### Device (please complete the following information):
 - OS: Windows 10
 - Go version: [e.g. 1.14]
 - Fyne version: latest

**Comment by andydotxyz** [comment]
This is very strange, I cannot figure it out at a glance. We seem to have a challenge with top-level widgets and re-theming.

You can work around it for now using
```
myWindow.SetContent(widget.NewVBox(form))
```

**Comment by andydotxyz** [comment]
There is now a fix on `release/v1.2.4` branch for testing, will be released next week
